curvefs/src/client/rpcclient/mds_client.cpp

```diff
@@ -306,16 +302,114 @@ bool MdsClientImpl::GetMetaServerListInCopysets(
             }
             cpinfoVec->push_back(copysetseverl);
         }
-        int ret = response.statuscode();
-        LOG_IF(WARNING, ret != 0) << "GetMetaServerList failed"
-                                  << ", errocde = " << response.statuscode()
-                                  << ", log id = " << cntl->log_id();
+        TopoStatusCode ret = response.statuscode();
+        LOG_IF(WARNING, TopoStatusCode::TOPO_OK != 0)
+            << "GetMetaServerList failed"
+            << ", errocde = " << response.statuscode()
+            << ", log id = " << cntl->log_id();
         return ret;
     };
 
     return 0 == rpcexcutor_.DoRPCTask(task, mdsOpt_.mdsMaxRetryMS);
 }
 
+bool MdsClientImpl::CreatePartition(
+    uint32_t fsID, uint32_t count, std::vector<PartitionInfo> *partitionInfos) {
+    auto task = RPCTask {
+        CreatePartitionResponse response;
+        mdsbasecli_->CreatePartition(fsID, count, &response, cntl, channel);
+        if (cntl->Failed()) {
+            LOG(WARNING) << "CreatePartition from mds failed, error is "
+                         << cntl->ErrorText()
+                         << ", log id = " << cntl->log_id();
+            return -cntl->ErrorCode();
+        }
+
+        TopoStatusCode ret = response.statuscode();
+        if (ret != TopoStatusCode::TOPO_OK) {
+            LOG(WARNING) << "CreatePartition: fsID = " << fsID
+                         << ", count = " << count << ", errcode = " << ret
+                         << ", errmsg = " << TopoStatusCode_Name(ret);
+            return ret;
+        }
+
+        partitionInfos->clear();
+        int partitionNum = response.partitioninfolist_size();
+        for (int i = 0; i < partitionNum; i++) {
```

Comment:
if response.partitioninfolist_size()==0, should return -1 and retry

Reply:
done